op-service/txplan/txplan.go

```diff
@@ -143,6 +161,10 @@ func WithType(t uint8) Option {
 
 func WithGasLimit(limit uint64) Option {
 	return func(tx *PlannedTx) {
+		// The gas limit is explicitly set so remove any dependencies which may have been added by a previous call
+		// to WithEstimator.
+		tx.Gas.ResetDependencies()
+		tx.Gas.Fn(nil)
```

Comment:
(nit) Looks like these are related calls - maybe we should combine them with something like:
```suggestion
		tx.Gas.ResetFn()
```

Reply:
Yeah makes sense - I named it `ResetFnAndDependencies` to be really clear. It's a bit awkward that dependencies and the function that uses them are registered separately to be honest, but changing that is outside the scope of this PR (and not particularly straightforward).